### ISSUE TITLE
JMS Connector does not ACK messages properly

### ISSUE THREAD

**Issue by fkoehler** [akka/alpakka#908]
Latest macOS, JDK 1.8.141, Alpakka: current master

When writing our own app I tried to write a simple test which puts messages on the queue, consumes them by ack'ing them via `ackSource`, after that I put more messages on the queue but when I consume the queue the old messages still show up. I expected that only the new messages show up since I acknowledged the other two. Adding `Thread.sleep` in between does not help. I added a test case to alpakka master to show the behaviour: https://github.com/fkoehler/alpakka/commit/cc0b2e5aa4573a1eb85462dcf0ab33560579c345

Is this a bug or do I have a general misconception on how all this is supposed to work?

**Comment by fkoehler** [comment]
Not sure, but reading through the issues this one seems to be related: #857

**Comment by fkoehler** [comment]
I looked into this in more detail and tried to reproduce the issue with the plain Java driver and that is possible as well. So it looks like this is not an Alpakka related issue but some, for me weird, behaviour inside the Java driver. I will look into this in more detail tonight.

**Comment by ennru** [comment]
Thank you for investigating further.

**Comment by fkoehler** [comment]
Investigating further we found that everyhing works as expected. Things seem to get a bit weird once two consumers are active but these seem to be JMS specific behaviour. 

We found one other interesting thing in the test and verified it in the implementation: In order to apply the backpressure with your own `ackQueue` messages are only acked when there is a new message coming in. This means the ack always lags behind one message. We understand the design decision though and are fine with it.

So I think we can close this.
